Removed retrieval of SIAModels from submodule, use Maven instead.

pascalalfadian opened the pull request:
Halo @davidatz285 & @harrysenjaya,

Bisa bantu review PR ini? Jadi IFStudentPortal yang sebelumnya itu, pakai metode `git submodule` untuk SIAModels nya. Tapi jadi masalah di heroku karena jadi ribet / tidak bisa untuk tarik submodule nya. Di branch baru ini, saya sudah menggunakan Maven sebagai gantinya.

Masalahnya, di Maven itu menggunakan SIAModels versi terbaru, yang _mungkin_ tidak kompatibel dengan IFStudentPortal.

Nah, bisakah kalian coba tes:

1. Kalau bisa di direktori baru, jangan dicampur dengan yang lama. Pull dan switch ke branch ini.
2. Jalankan aplikasinya seperti biasa
3. Coba login dan akses seperti biasa, apakah ada error?

Mohon infokan ya, berhasil / tidak nya. Karena saya tidak punya akun mahasiswa.

pascalalfadian committed:
Removed retrieval of SIAModels from submodule, use Maven instead.

davidatz285 commented:
Saya sudah mencoba run : 
1. Aplikasi jalan seperti biasa.
2. Login dan akses seperti biasa tidak ada error.

davidatz285 commented:
Update : 
1. Di poin 2 komentar saya sebelumnya tidak ada error di website, tapi di CMD ada exception java String index out of range di thread.
2. Jika mencoba login di jam-jam padat, saat koneksi time out akan ada error di website IF StudPor karena tidak ada respon dari student portal.

pascalalfadian commented:
Trims ya @davidatz285! 

> Di poin 2 komentar saya sebelumnya tidak ada error di website, tapi di CMD ada exception java String index out of range di thread.

Kalau sempat, boleh tolong check & fix problem di atas? Just to make sure tidak ada data yang ngaco gara2 exception itu. Atau alternatif nya, share password student kamu ke saya, hehe..

> Jika mencoba login di jam-jam padat, saat koneksi time out akan ada error di website IF StudPor karena tidak ada respon dari student portal.

Ini kalau mau di fix boleh, nggak juga gapapa sih mestinya. Bisa menyusul kapan2.

harrysenjaya commented:
Maaf baru mengabari Pa.

Barusan saya coba dan terjadi error yang mungkin dikarenakan login di jam yang padat. Namun setelah saya coba refresh dan melakukan login ulang, tidak ada masalah yang terjadi. Namun yang bisa diakses hanya menu "home", dan "info & lapor bug", menu lainnya terdapat error "String index out of range".

pascalalfadian commented:
Trims juga @harrysenjaya. Saya akan coba troubleshoot dulu.

pascalalfadian committed:
Fix for Persiapan Perwalian String index out of bounds error

Previously it is hardcoded to get the 11th script in the HTML code.
Now it iterates all scripts available.

pascalalfadian committed:
Jadwal kuliah moved to a new URL

pascalalfadian commented:
Kayaknya ini penyebabnya karena perubahan di Student Portal nya. Yang saya perbaiki sebenarnya `Scraper` nya hehe.. Tolong ditest lagi ya @davidatz285 & @harrysenjaya.

Kalau paling tidak di @davidatz285 sudah bisa jalan, minta tolong ya @davidatz285 untuk:

1. Merge PR ini
2. Boleh reset kembali password kamu.

Dan kalau PR ini sudah merged, silakan masing-masing mengambil update terbaru dari repo ini.

Trims.

davidatz285 commented:
Saya menemukan : 
1. Untuk halaman jadwal kuliah sudah bisa menampilkan jadwal kuliah kembali.
2. Untuk halaman syarat lulus dan persiapan perwalian, masih mengeluarkan exception String index out of bound di CMD.

harrysenjaya commented:
Setelah saya coba, semua halaman sudah dapat diakses dengan normal.

pascalalfadian commented:
Trims both!

@davidatz285 ini saya coba pakai akun kamu, kedua page tersebut tidak masalah: 
[Screen Shot 2020-10-08 at 14 42 45](https://user-images.githubusercontent.com/5976057/95429484-ab50fd00-0974-11eb-9bc7-42a5a07400cb.png) dan [Screen Shot 2020-10-08 at 14 43 00](https://user-images.githubusercontent.com/5976057/95429506-b0ae4780-0974-11eb-8ec9-1631522d380c.png). Mungkin bisa dicoba lagi di luar jam sibuk? Soalnya terkadang memang kalau session nya sudah, ini memang nggak ngasih tau, tau2 ada exception aja.

davidatz285 commented:
Kalau begitu saya merge PR ini ya Pak. Terima kasih.

davidatz285 commented:
Update : ternyata saya tidak punya write access, tidak bisa merge PR ini Pak maaf.

pascalalfadian commented:
Ups. OK saya yang merge.
